src/test/java/unit/GcsObjectReferenced.java

```diff
@@ -153,6 +153,110 @@ void listDescribeReflectAdd() throws IOException {
+    // --object-name=$objectName`
+    String name = "addRefToGcsFolder";
+    String folder = "blobs/";
+    UFGcsObject addedBucketObjectReference =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsObject.class,
+            "resource",
+            "add-ref",
+            "gcs-object",
+            "--name=" + name,
+            "--bucket-name=" + externalBucket.getName(),
+            "--object-name=" + folder);
+
+    // check that the name and bucket name match
+    assertEquals(name, addedBucketObjectReference.name, "add ref output matches name");
+    assertEquals(
+        externalBucket.getName(),
+        addedBucketObjectReference.bucketName,
+        "add ref output matches bucket name");
+    assertEquals(
+        folder, addedBucketObjectReference.objectName, "add ref output matches bucket object name");
+
+    // check that the object is in the list
+    List<UFGcsObject> matchedResourceList = listObjectResourceWithName(name);
+    assertEquals(1, matchedResourceList.size());
+    UFGcsObject matchedResource = matchedResourceList.get(0);
+    assertEquals(name, matchedResource.name, "list output matches name");
+    assertEquals(
+        externalBucket.getName(), matchedResource.bucketName, "list output matches bucket name");
+    assertEquals(folder, matchedResource.objectName, "List output matches bucket object name");
+
+    // `terra resource describe --name=$name --format=json`
+    UFGcsObject describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsObject.class, "resource", "describe", "--name=" + name);
+
+    // check that the name and bucket name match
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertEquals(
+        externalBucket.getName(),
+        describeResource.bucketName,
+        "describe resource output matches bucket name");
+    assertEquals(
+        folder, describeResource.objectName, "describe resource output matches bucket object name");
+
+    String name2 = "addReftoGcsFilesInFolder";
+    String filesInFolder = "blobs/*";
+    UFGcsObject addedBucketObjectReference2 =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsObject.class,
+            "resource",
+            "add-ref",
+            "gcs-object",
+            "--name=" + name2,
+            "--bucket-name=" + externalBucket.getName(),
+            "--object-name=" + filesInFolder);
+    // check that the name and bucket name match
+    assertEquals(name2, addedBucketObjectReference2.name, "add ref output matches name");
+    assertEquals(
+        externalBucket.getName(),
+        addedBucketObjectReference2.bucketName,
+        "add ref output matches bucket name");
+    assertEquals(
+        filesInFolder,
+        addedBucketObjectReference2.objectName,
+        "add ref output matches bucket object name");
+
+    String name3 = "addRefToGcsTextFilesInFolder";
+    String textFilesInFolder = "blobs/*.txt";
```

Comment:
it wouldn't fail actually because the user have access to everything in the blobs/ folder. 

Currently in WSM the check access code use something like `storage.get(bucketname, blobId)` to get the blob. If get succeeded and no exception is thrown, we know the user have at least READER access. However, if the blobId is "blobs/**", "blobs/*.txt", "blobs/*.png", there's actually no such object in the storage, so storage.get() will succeed but will return null. Currently checkAccess will still pass for these cases to be more flexible on what to pass in.

Reply:
Would "blobs1234/" fail since that folder doesn't exist? Or can you give an example that should fail?

Reply:
i believe it won't fail either because this log-in user has bucket level access so they can perform get action to anything in this bucket. 

I set the check very lenient because Nichole is interested in having a reference to foo/**, foo/**.txt, etc which are not actual objects but we can tell the access level roughly by checking if the user can perform get action inside folder foo/ regardless of whether there's anything in that folder. We could make the check stricter on the WSM side but I think that's outside the scope of this PR.